packages/next/client/link.tsx

```diff
@@ -55,7 +55,7 @@ export type LinkProps = {
 }
 
 let observer: IntersectionObserver
-const listeners = new Map()
+const listeners = new Map<Element, () => void>()
 const IntersectionObserver =
   typeof window !== 'undefined' ? (window as any).IntersectionObserver : null
```

Comment:
I think you can just do `window.IntersectionObserver` here and then drop the type annotation on `entries` below.

Reply:
Good idea! I'm changing